v2/tests/log_internal_error_ut/log_internal_error_ut.c

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <stdbool.h>
+#if defined _MSC_VER
+#include <stdlib.h>
+#endif
+
+#include "macro_utils/macro_utils.h"
+#include "c_logging/log_internal_error.h"
+
+static size_t asserts_failed = 0;
+
+// defines how many mock calls we can have
+#define MAX_MOCK_CALL_COUNT (128)
+
+#define MOCK_CALL_TYPE_VALUES \
+    MOCK_CALL_TYPE_abort, \
+    MOCK_CALL_TYPE__set_abort_behavior \
+
+MU_DEFINE_ENUM(MOCK_CALL_TYPE, MOCK_CALL_TYPE_VALUES)
+
+// very poor mans mocks :-(
+typedef struct _set_abort_behavior_CALL_TAG
+{
+    unsigned int _Flags;
+    unsigned int _Mask;
+} _set_abort_behavior_CALL;
+
+typedef struct MOCK_CALL_TAG
+{
+    MOCK_CALL_TYPE mock_call_type;
+    union
+    {
+        _set_abort_behavior_CALL _set_abort_behavior_call;
+    } u;
```

Comment:
>u; [](http://example.com/codeflow?start=6&length=2)

if "u" is missing - from here and everywhere you can just dive straight into the fields of the union.

Reply:
Got it, obviously it would produce a warning on our compiler also.
I can leave with it as is for now.